frontend/src/components/transactions/Transactions.tsx

```diff
@@ -64,28 +73,70 @@ export default class extends React.Component<Props, State> {
     if (this._transactionsApi === null) {
       this._transactionsApi = new TransactionsApi();
     }
-    const transactions = await this._transactionsApi.getTransactions({
+    const transactions = (await this._transactionsApi.getTransactions({
       signerId: this.props.accountId,
       receiverId: this.props.accountId,
       blockHash: this.props.blockHash,
       tail: this.props.reversed,
-      limit: this.props.limit
-    });
+      limit: this.state.stop
+    })) as T.Transaction[];
     this.setState({ transactions });
   };
 
+  _isAtBottom = () => {
+    return (
+      this._transactionLoader &&
+      this._transactionLoader.getBoundingClientRect().bottom <=
+        window.innerHeight
+    );
+  };
+
+  _onScroll = async () => {
+    this._transactionLoader = document.querySelector("#tx");
+    const bottom = this._isAtBottom();
+    if (bottom) {
+      document.removeEventListener("scroll", this._onScroll);
+      await this._loadTransactions();
+      document.addEventListener("scroll", this._onScroll);
+    }
+  };
+
+  _loadTransactions = async () => {
+    this.setState({ stop: this.state.stop + this.props.limit, loading: true });
```

Comment:
`stop` variable raised some questioning in my mind. At first, I just didn't like the naming, but on the second thought I realized that with the current design it would be much cleaner just using `this.state.transactions.length` and query for `this.state.transactions.length + this.props.limit`.

Another naming item is `this.props.limit`. It looses its semantic in this case, so we should rename it to `paginationSize` or something.

**The following is a valid concern (I wrote it before I realized the depth of the issue), but irrelevant given the above.**

`setState` function is not a sync function, so you should never assume:
1. that `this.state` is up to date (pass a function into `setState`, which should expect the previous state and produce a new one:

```js
this.setState(({ stop }) => ({ stop: stop + this.props.limit, loading: true }))
```
2. that consequent `setState` will be executed in the right order (in theory, the latter `loading: false` may end up executed before this `loading: true`), so you should use a callback as a second argument to the first `setState`. I suggest [creating a promise there](https://stackoverflow.com/a/47019631/1178806) and awaiting for it and `fetchTransaction` with `await Promise.all([...])` and only then call to `setState({ loading: false })`